### ISSUE TITLE
Advice on gradient clipping

### ISSUE THREAD

**Issue by willprice** [pytorch/ignite#375]
Using `ignite.engine.create_supervised_trainer` I'm unable to clip my gradients as this falls between the steps:
```python
loss.backward()
# gradient clip
optimizer.step()
```
There are no event's fired between these two pieces of code. Gradient clipping and other manipulation techniques are quite common. It'd be nice if Ignite supported these. Is there a recommended way of doing this? Or is it just a case of constructing my own `Engine` with a process_fn similar to that constructed by `create_supervised_trainer`? Are there any plans to create events that allow one to do this sort of thing with `create_supervised_trainer`?

Cheers for the awesome lib!

**Comment by vfdev-5** [comment]
@willprice thanks for the feedback! 

Concering your problem we advise (see [Concepts](https://pytorch.org/ignite/concepts.html)) to use directly `Engine` with custom `process_fn` without complicating the system with custom events. 
Methods like `create_supervised_trainer` are just helper methods for a basic usage.

HTH

**Comment by willprice** [comment]
Hi @vfdev-5,
Thanks for the quick response and advice, makes sense!
